src/blobs/BlobRegistry.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import "./IBlobRegistry.sol";
+
+/// @title BlobRegistry
+contract BlobRegistry is IBlobRegistry {
+    /// @dev A mapping of the hash of a blob source to the timestamp when it was saved
+    mapping(bytes32 sourceHash => uint256 timestamp) private _savedHashes;
+
+    /// @inheritdoc IBlobRegistry
+    function getAndSaveHash(uint256[] calldata blobIdxs)
```

Comment:
If we assume there are not going to be more than 255 blobs, then it's possible to express this array as a `bytes32 blobIdxs` bitmap so that we don't allocate calldata for the indexes. For example, instead of `[0,1,2]` we can provide `0x111000...`

Reply:
Previously I did have another function `function getAndSaveHash(bytes32 encodedBlobIndices)` but I deleted as I don't know if we need it here or a calling dapp shall encode the indices in its function.

Reply:
I understand the dapp should encode the indices on their side, so I agree it's not ideal. Current Implementation is good.

Reply:
See https://github.com/OpenZeppelin/minimal-rollup/pull/33